x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPlugin.java

```diff
@@ -138,16 +139,52 @@ public class EnrichPlugin extends Plugin implements SystemIndexPlugin, IngestPlu
         Setting.Property.NodeScope
     );
 
+    /**
+     * This setting solely exists because the original setting was accidentally renamed in
+     * https://github.com/elastic/elasticsearch/pull/111412.
+     */
+    @UpdateForV10(owner = UpdateForV10.Owner.DATA_MANAGEMENT)
+    public static final String CACHE_SIZE_SETTING_BWC_NAME = "enrich.cache.size";
+    public static final Setting<FlatNumberOrByteSizeValue> CACHE_SIZE_BWC = new Setting<>(
+        CACHE_SIZE_SETTING_BWC_NAME,
+        (String) null,
+        (String s) -> FlatNumberOrByteSizeValue.parse(
+            s,
+            CACHE_SIZE_SETTING_BWC_NAME,
+            new FlatNumberOrByteSizeValue(ByteSizeValue.ofBytes((long) (0.01 * JvmInfo.jvmInfo().getConfiguredMaxHeapSize())))
+        ),
+        Setting.Property.NodeScope,
+        Setting.Property.Deprecated
+    );
+
     private final Settings settings;
     private final EnrichCache enrichCache;
+    private final long maxCacheSize;
 
     public EnrichPlugin(final Settings settings) {
         this.settings = settings;
-        FlatNumberOrByteSizeValue maxSize = CACHE_SIZE.get(settings);
+        FlatNumberOrByteSizeValue maxSize;
+        if (settings.hasValue(CACHE_SIZE_SETTING_BWC_NAME)) {
+            if (settings.hasValue(CACHE_SIZE_SETTING_NAME)) {
+                throw new IllegalArgumentException(
+                    Strings.format(
+                        "Both [{}] and [{}] are set, please use [{}]",
+                        CACHE_SIZE_SETTING_NAME,
+                        CACHE_SIZE_SETTING_BWC_NAME,
+                        CACHE_SIZE_SETTING_NAME
+                    )
+                );
```

Comment:
I worry that by blocking this we kill the ES node on startup, do you think it'd be better to log this at ERROR level so that we don't block the startup in the case of a user doing an automated upgrade?

Reply:
That was kind of my intention here. These settings can only be set in the file-based settings, are node-specific, and there's no automated mechanism that updates the setting name. Therefore, I figure this exception will only be thrown if a user has _manually_ altered the `elasticsearch.yml` on a specific node, meaning they're already altering their settings manually so we might as well tell them immediately they're doing something wrong - rather than logging an error they can choose to ignore and bump into again when they upgrade to some version down the line. What do you think?

Reply:
I am thinking of someone doing their own version of ES-as-a-service that wants to set this setting, they have one default es.yml file that works regardless of ES version (sort of if they had a docker image with different ES versions and one config dir they mounted everywhere). We would preclude that if we made them mutually exclusive, and they'd have to substitute in different values depending on which 8.x release they were using.